content/docs/cucumber/state.md

```diff
@@ -440,22 +488,27 @@ Feature: Let's write a lot of stuff to the DB
 ```
 
 ### With JUnit and Spring
-
+{{% block "java,kotlin" %}} 
 See the [`spring-txn`](https://github.com/cucumber/cucumber-jvm/tree/master/examples/spring-txn) example in Cucumber-JVM for a minimal setup.
+{{% /block %}}
+
+{{% block "ruby,javascript" %}} JUnit and Spring are used with JVM languages. {{% /block %}}
 
 # Browser Automation and Transactions
 
-If you're using a [browser automation](/docs/guides/browser-automation) tool that talks to your application over HTTP, the
-transactional approach will not work if your [step definitions](/docs/cucumber/step-definitions) and the web application serving
-HTTP request each have their own database connection.
-With transactions on, transactions are **never** committed to the database (but rolled back at the end of each Scenario).
-Therefore, the web server's connection will never see data from Cucumber, and therefore your browser won't either.
-Likewise, Cucumber's connection won't see data from the web server.
+If you're using a [browser automation](/docs/guides/browser-automation) tool that talks to your application over HTTP,
+the transactional approach will not work if your [step definitions](/docs/cucumber/step-definitions) and the web
+application serving HTTP request each have their own database connection. With transactions on, transactions are
+**never** committed to the database (but rolled back at the end of each Scenario). Therefore, the web server's
```

Comment:
I'm sure this makes more sense to experienced coders. Let me see if I get it. If I am using a browser automation tool like WebDriver, I have to *only* use the database connection from that (WebDriver), I cannot try to do any database updates from Cucumber step definitions, because they will never be persisted? I have no idea how a Selenium WebDriver script gets connected to Cucumber step definitions. Through the automation code?

I would add the word "are" to the phrase in parens - "(but are rolled back at the end of each scenario)"

This seems quite confusing to me, but again, since I have zero experience with this, that's probably normal. Is it worth having an example, or people will just understand this?

Reply:
Yes, "transaction" here is a database transaction. This section describes that if your front end AND your cucumber code use DIFFERENT connections to the database, they might not see each others data if the transaction is rolled back. Does that make sense? I'm wondering whether we could describe this more clearly.

Reply:
An example would help me, but it is probably unnecessary for the average Cucumber coder.